test/components/tree-view.spec.tsx

```diff
@@ -123,18 +123,223 @@ describe('<TreeView />', () => {
             const onSelectItem = sinon.spy();
             const { select } = clientRenderer.render(<TreeView dataSource={treeData} onSelectItem={onSelectItem}/>);
 
-            const nodeLabelToClick = getTreeItem(treeData[0].label);
+            selectItemWithLabel(select, treeData[0].label);
+
+            return waitFor(() => expect(onSelectItem).to.have.been.calledWithMatch(treeData[0]));
+        });
+
+        describe('Keyboard Navigation', () => {
+            it('expands and collapses focused treeItem when right and left arrows are clicked', async () => {
+                const { select, waitForDom } = clientRenderer.render(<TreeViewDemo />);
+
+                const nodeChildren = treeData[0].children;
+
+                selectItemWithLabel(select, treeData[0].label);
+                expandItemWithLabel(select, treeData[0].label);
+
+                await waitForDom(() => expect(select(getTreeItem(nodeChildren![1].label))).to.be.present());
+
+                simulate.keyDown(select('TREE_VIEW_DEMO', 'TREE_VIEW'), { keyCode: KeyCodes.LEFT });
+                await waitForDom(() => expect(select(getTreeItem(nodeChildren![1].label))).to.be.absent());
+
+                simulate.keyDown(select('TREE_VIEW_DEMO', 'TREE_VIEW'), { keyCode: KeyCodes.RIGHT });
+
+                return waitForDom(() => expect(select(getTreeItem(nodeChildren![1].label))).to.be.present());
+            });
+
+            it('returns to parent if there is after collapsing the element if possible when left is clicked', async () => {
+                const { select, waitForDom } = clientRenderer.render(<TreeViewDemo />);
+
+                const nodeChildren = treeData[0].children;
+
+                expandItemWithLabel(select, treeData[0].label);
+
+                await waitForDom(() => expect(select(getTreeItem(nodeChildren![1].label))).to.be.present());
+
+                selectItemWithLabel(select, nodeChildren![1].label);
+
+                await waitForDom(() => expect(select(getTreeItem(nodeChildren![1].label))).to.have.attr('data-focused', 'true'));
 
-            simulate.click(select(nodeLabelToClick));
+                simulate.keyDown(select('TREE_VIEW_DEMO', 'TREE_VIEW'), { keyCode: KeyCodes.LEFT });
 
-            return waitFor(() => {
-                expect(onSelectItem).to.have.been.calledOnce;
-                expect(onSelectItem).to.have.been.calledWithMatch(treeData[0]);
+                return waitForDom(() => expect(select(getTreeItem(treeData[0].label))).to.have.attr('data-focused', 'true'));
+            });
+
+            it('moves to child to if there is one after expanding the element if possible when right is clicked', async () => {
```

Comment:
How about: "when the keyboard focus is on an expanded parent, pressing the right arrow key moves the focus to its first child"?